[custom_ops] Block overload names

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #106088
* #106076
* __->__ #106075

These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

zou3519 committed:
[custom_ops] Block overload names

These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_ops] Block overload names"


These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_ops] Block overload names"


These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

[ghstack-poisoned]

soulitzer commented on torch/_custom_op/impl.py:
```diff
@@ -561,10 +561,16 @@ def validate_schema(schema: FunctionSchema) -> None:
+        raise ValueError(f"The torch.custom_ops APIs do not handle non-default "
+                         f"overloads, i.e. operator names with '.' in them. "
+                         f"Please name your operator something like ns::foo. "
+                         f"Got: {qualname}")
```
nit: could users be confused about what a "non-default overload" is in this context? My first thought is that that would mean default overload must necessarily end with `.default`, but then as I read further I do see that we clarify that we don't allow `.` in the name, but maybe "do not handle more than one overload" better corresponds to "don't allow periods".

zou3519 committed:
Update on "[custom_ops] Block overload names"


These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_ops] Block overload names"


These are valid with the torch.library API, but (1) they add complexity
and (2) I have never seen a custom op actually use an overload name
before. For simplicity we block all overloads.

Test Plan:
- new test

[ghstack-poisoned]
